sensorhub-core/src/main/java/org/sensorhub/impl/comm/TCPServerCommProvider.java

```diff
@@ -0,0 +1,136 @@
+/***************************** BEGIN LICENSE BLOCK ***************************
+ The contents of this file are subject to the Mozilla Public License, v. 2.0.
+ If a copy of the MPL was not distributed with this file, You can obtain one
+ at http://mozilla.org/MPL/2.0/.
+
+ Software distributed under the License is distributed on an "AS IS" basis,
+ WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
+ for the specific language governing rights and limitations under the License.
+
+ Copyright (C) 2023 Botts Innovative Research Inc. All Rights Reserved.
+ ******************************* END LICENSE BLOCK ***************************/
+
+package org.sensorhub.impl.comm;
+
+import org.sensorhub.api.comm.ICommProvider;
+import org.sensorhub.api.common.SensorHubException;
+import org.sensorhub.impl.module.AbstractModule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ssl.SSLServerSocket;
+import javax.net.ssl.SSLServerSocketFactory;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.ServerSocket;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+
+/**
+ * Communication provider for TCP/IP server connections.
+ *
+ * @author Michael Elmore
+ * @since September 2023
+ */
+public class TCPServerCommProvider extends AbstractModule<TCPServerCommProviderConfig> implements ICommProvider<TCPServerCommProviderConfig> {
+    static final Logger log = LoggerFactory.getLogger(TCPServerCommProvider.class.getSimpleName());
+    private final Set<Consumer<ConnectionEventArgs>> eventHandlers = new HashSet<>();
+
+    ServerSocket serverSocket;
+    Thread serverThread;
+
+    public TCPServerCommProvider() {
+    }
+
+    @Override
+    protected void doStart() throws SensorHubException {
+        var config = this.config.protocol;
+
+        //Start a thread to listen on the configured port
+        serverThread = new Thread(() -> {
+            try {
+                if (config.enableTLS) {
+                    var sslServerSocketFactory = (SSLServerSocketFactory) SSLServerSocketFactory.getDefault();
+                    try (var sslServerSocket = (SSLServerSocket) sslServerSocketFactory.createServerSocket(config.localPort)) {
+                        while (true) {
```

Comment:
I would not use a loop like this, I would create a variable that is synchronized like an AtomicBoolean to control execution of the loop(s) (here and line 64). In the doStop method I would set that AtomicBoolean false so that the active loop in the thread knows when to exit. 

Interrupt causes an InterruptedException to be thrown by the thread.  As a general rule using interrupts are expensive computationally and when possible, should be avoided to control program logic and flow.  Exceptions should be exceptional.